src/tasks/AndroidAppBuilder/Templates/monodroid.c

```diff
@@ -33,8 +33,11 @@
 void
 Java_net_dot_MonoRunner_setEnv (JNIEnv* env, jobject thiz, jstring j_key, jstring j_value);
 
+void
+Java_net_dot_MonoRunner_initRuntime (JNIEnv* env, jobject thiz, jstring j_files_dir, jstring j_cache_dir, jstring j_testresults_dir, long current_local_time);
+
 int
-Java_net_dot_MonoRunner_initRuntime (JNIEnv* env, jobject thiz, jstring j_files_dir, jstring j_cache_dir, jstring j_testresults_dir, jstring j_entryPointLibName, jobjectArray j_args, long current_local_time);
```

Comment:
@steveisok does this affect librarymode? I don't really see how this works

Reply:
Library mode uses "on library load" runtime initialization and is built as part of the lib: https://github.com/dotnet/runtime/blob/0a477a86d0fea6234ada63e660c0faa310fe121d/src/tasks/LibraryBuilder/Templates/autoinit.c#L164 

and a separate monodroid source is used: https://github.com/dotnet/runtime/blob/0a477a86d0fea6234ada63e660c0faa310fe121d/src/tasks/AndroidAppBuilder/Templates/monodroid-librarymode.c 

so if we are changing the JNI method signature, that one ^ should also be updated.

@matouskozak I added a library mode test in: https://github.com/dotnet/runtime/pull/111545 and would put that PR also as a prerequisite to this one, so we have a better test coverage.

Finally, I would treat library mode as out of scope.